From d46305bc47f8e454e96030e3d20bde72da0527ef Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 14:38:19 -0400
Subject: [ethernet] enic: include irq.h for irqreturn_t definitions

Message-id: <1409668716-4814-8-git-send-email-sassmann@redhat.com>
Patchwork-id: 90633
O-Subject: [RHEL7.1 PATCH 07/24] net: enic: include irq.h for irqreturn_t definitions
Bugzilla: 1107934
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Josh Boyer <jwboyer@gmail.com>

The enic driver fails to build on ARM with:

In file included from drivers/net/ethernet/cisco/enic/enic_res.c:40:0:
drivers/net/ethernet/cisco/enic/enic.h:48:2: error: expected specifier-qualifier-list before 'irqreturn_t'
  irqreturn_t (*isr)(int, void *);
  ^

Nothing in the driver is explicitly including the irq definitions, so we add
an include of linux/irq.h to pick them up.

Signed-off-by: Josh Boyer <jwboyer@fedoraproject.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit fef1f07cbf3d9b6f79beac8575554476c02c7f54)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/cisco/enic/enic.h b/drivers/net/ethernet/cisco/enic/enic.h
index e9f7c65..e35c8e0 100644
--- a/drivers/net/ethernet/cisco/enic/enic.h
+++ b/drivers/net/ethernet/cisco/enic/enic.h
@@ -29,6 +29,7 @@
 #include "vnic_stats.h"
 #include "vnic_nic.h"
 #include "vnic_rss.h"
+#include <linux/irq.h>
 
 #define DRV_NAME  "enic"
 #define DRV_DESCRIPTION  "Cisco VIC Ethernet NIC Driver"
-- 
1.7.1